File: validation/reconstruction/validateReconstruction.ts
import { buildRuntimeGraph, reconstructRuntime } from "../../src/index.js";

const graph = buildRuntimeGraph({ a: 1 });
const extraction = { unified_runtime_graph: graph, graph };
const r1 = reconstructRuntime({ extraction });
const r2 = reconstructRuntime({ extraction });
const id1 = (r1.runtime as Record<string, unknown>).runtime_id;
const id2 = (r2.runtime as Record<string, unknown>).runtime_id;
const results = {
  reconstruction_match: id1 === id2,
  bounded: r1.bounded === true,
};
const allOk = Object.values(results).every(Boolean);
console.log(allOk ? "PASS" : "FAIL", results);
if (!allOk) process.exit(1);

File: validation/reconstruction/validate_reconstruction.dart
import 'dart:convert';
import 'dart:io';

import 'package:webweavex/webweavex.dart';

void main() {
  final graph = buildRuntimeGraph({'a': 1});
  final extraction = {
    'unified_runtime_graph': graph.toJson(),
    'graph': graph.toJson(),
  };
  final r1 = reconstructRuntimeFromEnvelope(extraction: extraction);
  final r2 = reconstructRuntimeFromEnvelope(extraction: extraction);

  final results = {
    'reconstruction_match': r1['runtime_id'] == r2['runtime_id'],
    'reconstructed': r1['reconstructed'] == true,
    'bounded': r1['bounded'] == true,
  };

  final allOk = results.values.every((x) => x == true);
  Directory('validation/reconstruction').createSync(recursive: true);
  File('validation/reconstruction/reconstruction_vectors.json')
      .writeAsStringSync(
    const JsonEncoder.withIndent('  ').convert({
      'algorithm': 'webweavex-reconstruction-v2.0.0',
      'vectors': [results],
    }),
  );
  stdout.writeln(allOk ? 'PASS' : 'FAIL');
  if (!allOk) exit(1);
}
